### PR TITLE
WSL Survey 20231212

### PR DESCRIPTION
**Pull request by Fearyncess**
Topic Description
-----------------

While AOSC OS has a WSL distro edition, it still have no adjustments for WSL specifically. This survey will introduce a series of packages, improving Out-Of-Box user experiences for WSL users.

Package(s) Affected
-------------------

**Added:** 
* wsl-base, wsl-default-settings, aosc-os-presets-wsl, wsl-drop-caches

Security Update?
----------------

No

Build Order
-----------

```
#buildit wsl-drop-caches wsl-default-settings aosc-os-presets-wsl wsl-base
```

Test Build(s) Done
------------------

**Primary Architectures**
- [x] Architecture-independent `noarch`

### CODE REVIEW

File: app-utils/wsl-drop-caches/spec
Comment (KexyBiscuit) [inline app-utils/wsl-drop-caches/spec:1]: Microsoft has introduced [an experimental feature](https://github.com/microsoft/WSL/issues/4166#issuecomment-1724487010) just like this, whether wait for the feature to be stabilized or not is a question.

File: runtime-data/wsl-default-settings/autobuild/build [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+
+abinfo "Writing default settings to wsl.conf ..."
+cat > "$PKGDIR"/etc/wsl.conf << EOF
+[boot]
+# Enable Systemd support
+systemd=true
+
+[automount]
+# Enable rootfs sharing between WSL distros support
+crossDistro = true
```
Comment (KexyBiscuit) [inline runtime-data/wsl-default-settings/autobuild/build]: This option seems to be [abandoned](https://github.com/microsoft/WSL/issues/4577#issuecomment-685173446).

**Review by KexyBiscuit — CHANGES_REQUESTED** [review]
LGTM overall, two bits here.

**Comment by Fearyncess** [comment]
WSL's memoryReclaim​ is a experimental feature, while some user wouldn't like to modify .wslconfig​ file in the host. So I think that made this helper by default is better, until Microsoft's devteam decides to land this feature.

After a test, filesystem sharing between WSL's distros works well w/o crossDistro​ option. I'll remove this line later.
________________________________
发件人: Kexy Biscuit ***@***.***>
发送时间: 2023年12月14日 9:22
收件人: AOSC-Dev/aosc-os-abbs ***@***.***>
抄送: Lain Yang ***@***.***>; Author ***@***.***>
主题: Re: [AOSC-Dev/aosc-os-abbs] WSL Survey 20231212 (PR #4904)


@KexyBiscuit requested changes on this pull request.

LGTM overall, two bits here.

________________________________

On app-utils/wsl-drop-caches/spec<https://github.com/AOSC-Dev/aosc-os-abbs/pull/4904#discussion_r1426057366>:

Microsoft has introduced an experimental feature<https://github.com/microsoft/WSL/issues/4166#issuecomment-1724487010> just like this, whether wait for the feature to be stabilized or not is a question.

________________________________

In runtime-data/wsl-default-settings/autobuild/build<https://github.com/AOSC-Dev/aosc-os-abbs/pull/4904#discussion_r1426059367>:

> @@ -0,0 +1,12 @@
+mkdir -pv "$PKGDIR"/etc
+
+abinfo "Writing default settings to wsl.conf ..."
+cat > "$PKGDIR"/etc/wsl.conf << EOF
+[boot]
+# Enable Systemd support
+systemd=true
+
+[automount]
+# Enable rootfs sharing between WSL distros support
+crossDistro = true


This option seems to be abandoned<https://github.com/microsoft/WSL/issues/4577#issuecomment-685173446>.

**Review by KexyBiscuit — DISMISSED** [review]
LGTM for now.

File: meta-bases/wsl-base/autobuild/build [hunk 2]
```diff
@@ -0,0 +1 @@
+mkdir "${PKGDIR}"
```
Comment (MingcongBai) [inline meta-bases/wsl-base/autobuild/build]: Use `ABTYPE=dummy`.

File: app-utils/wsl-drop-caches/spec [hunk 3]
```diff
@@ -0,0 +1,3 @@
+VER=0.4
+SRCS="tbl::https://github.com/arkane-systems/wsl-drop-cache/releases/download/v${VER}/wsl-drop-caches_${VER}.tar.xz"
+CHKSUMS="sha256::d4a01b0aee63a7ad9fc2fa6b5514ffd364ce33e403c8abaf2e1374f2f5a0c912"
```
Comment (MingcongBai) [inline app-utils/wsl-drop-caches/spec:3]: Missing CHKUPDATE.

File: app-utils/wsl-drop-caches/autobuild/defines [hunk 4]
```diff
@@ -0,0 +1,5 @@
+PKGNAME=wsl-drop-caches
+PKGSEC=misc
```
Comment (MingcongBai) [inline app-utils/wsl-drop-caches/autobuild/defines]: utils

File: app-utils/wsl-drop-caches/autobuild/defines [hunk 5]
```diff
@@ -0,0 +1,5 @@
...
+PKGDES="A tool reclaiming cached memory in WSL2 periodically if idle"
```
Comment (MingcongBai) [inline app-utils/wsl-drop-caches/autobuild/defines]: A tool for reclaiming idle cached memory periodically in WSL2.

**Review by MingcongBai — CHANGES_REQUESTED** [review]

File: app-utils/wsl-drop-caches/autobuild/defines (hunk 5 above)
Comment (Fearyncess) [reply app-utils/wsl-drop-caches/autobuild/defines]: descr. editted

File: app-utils/wsl-drop-caches/autobuild/defines (hunk 4 above)
Comment (Fearyncess) [reply app-utils/wsl-drop-caches/autobuild/defines]: section changed

File: meta-bases/wsl-base/autobuild/build (hunk 2 above)
Comment (Fearyncess) [reply meta-bases/wsl-base/autobuild/build]: ABTYPE changed

File: app-utils/wsl-drop-caches/autobuild/build [hunk 6]
```diff
@@ -0,0 +1,5 @@
+abinfo "Installing binaries..."
+
+install -Dm755 -t "${PKGDIR}/usr/bin" "${SRCDIR}/drop_cache_if_idle"
```
Comment (MingcongBai) [inline app-utils/wsl-drop-caches/autobuild/build]: Verbosity.

File: app-utils/wsl-drop-caches/autobuild/defines [hunk 7]
```diff
@@ -0,0 +1,5 @@
+PKGNAME=wsl-drop-caches
+PKGSEC=utils
+PKGDES="A tool for reclaiming idle cached memory periodically in WSL2."
```
Comment (MingcongBai) [inline app-utils/wsl-drop-caches/autobuild/defines]: Remove the period point at the end.

**Review by MingcongBai — CHANGES_REQUESTED** [review]

**Review by KexyBiscuit — APPROVED** [review]

**Review by MingcongBai — APPROVED** [review]

**Pull request merged**